09-clustering.Rmd

```diff
@@ -148,79 +152,83 @@ In K-means clustering, we measure the quality of a cluster by its
 base <- ggplot(clus1) +
-  geom_point(aes(y = loyalty, x = csat)) +
-  labs(x = "Customer satisfaction", y = "Loyalty") +
-  xlim(c(0, 4)) +
-  ylim(c(6, 9)) +
+  geom_point(aes(y = bill_length_scaled, x = flipper_length_scaled)) +
+  labs(x = "Flipper Length (scaled)", y = "Bill Length (scaled)") +
   theme(legend.position = "none")
 
-mn <- clus1 %>% summarize(csat = mean(csat), loyalty=mean(loyalty))
-for (i in 1:5) {
-  base <- base + geom_segment(x = unlist(mn[1, "csat"]), y = unlist(mn[1, "loyalty"]),
-                           xend = unlist(clus1[i, "csat"]), yend = unlist(clus1[i, "loyalty"]))
+mn <- clus1 %>% summarize(flipper_length_scaled = mean(flipper_length_scaled), bill_length_scaled = mean(bill_length_scaled))
+for (i in 1:nrow(clus1)) {
+  base <- base + geom_segment(
+    x = unlist(mn[1, "flipper_length_scaled"]), y = unlist(mn[1, "bill_length_scaled"]),
+    xend = unlist(clus1[i, "flipper_length_scaled"]), yend = unlist(clus1[i, "bill_length_scaled"])
+  )
 }
-base <- base + geom_point(aes(y = mean(loyalty), x = mean(csat), colour="red", size=2.5)) 
+base <- base + geom_point(aes(y = mean(bill_length_scaled), x = mean(flipper_length_scaled), colour = "red", size = 2.5))
 base
 ```
 
-The larger the value of $S^2$, the more spread-out the cluster is, since large $S^2$ means that points are far away from the cluster center.
-Note, however, that "large" is relative to *both* the scale of the variables for clustering *and* the number of points in the cluster; a 
-cluster where points are very close to the center might still have a large $S^2$ if there are many data points in the cluster. 
+The larger the value of $S^2$, the more spread-out the cluster is, since large $S^2$ means that points are far from the cluster center.
+Note, however, that "large" is relative to *both* the scale of the variables for clustering *and* the number of points in the cluster. A cluster where points are very close to the center might still have a large $S^2$ if there are many data points in the cluster. 
 
 ### The clustering algorithm
 
-The K-means algorithm is quite simple. We begin by picking K, and uniformly randomly assigning data to the K clusters. 
+We begin the K-means algorithm by picking K, and uniformly randomly assigning data to the K clusters. 
 Then K-means consists of two major steps that attempt to minimize the
 sum of WSSDs over all the clusters, i.e. the *total WSSD*:
 
 1. **Center update:** Compute the center of each cluster.
 2. **Label update:** Reassign each data point to the cluster with the nearest center.
 
 These two steps are repeated until the cluster assignments no longer change.
-For example, in the customer data example from earlier, our initialization might look like this:
+For example, in the penguin data example, our initialization might look like this:
```

Comment:
i like the color scheme, but maybe it's not great for colorblind vision? pass these images through a colorblind filter to see what they look like, and if needed change to a different scheme